### ISSUE TITLE
sophisticated backpacks not dye-able

### ISSUE THREAD

**Issue by caelansimms** [TerraFirmaGreg-Team/Modpack-1.20.x#364]
### Version modpack Used

0.7.8

### Environment

Singleplayer

### New Worlds

Yes

### Expected Behavior

Backpacks supplied from SophisticatedBackpacks Mod are dye-able, involving the body of the backpack and the straps of the backpack, allowing for 2 colors to be added on

### Actual Behavior

Backpacks are not able to be dyed with any dye inside a crafting menu, 2x2 or 3x3.

### Steps to Reproduce

1. Craft/grab a basic backpack (tier 1 or otherwise)
2. Select any dye and attempt to dye the backpack by placing a dye/dyes inside the grid alongside the backpack 

### Additional Information

This _is_ a feature inside other versions and other modpacks, with no addons needed to make this possible. I highly doubt that it was removed. I can only suspect in the process of reworking the backpacks outside of "Backpack Tier X" catagories, substituted with the present Backpack, Red Steel Backpack, Titanium Backpack, etc. that this is where the hiccup originates.
Cheers.

**Comment by Exzept1on** [comment]
Works right.

**Comment by Exzept1on** [comment]
Thanks for shit dev, who can't fix bugs or problems with his mod.

**Comment by Exzept1on** [comment]
Maybe we will fix this by install other backpack mod

**Comment by caelansimms** [comment]
Okay, didn’t know it was a mod-specific bug. 

Personally, this is IMO the best backpack mod going around & the bug is purely cosmetic so I can live with it, I request we don’t change it.

thanks for replying!

**Comment by BabBimmaBome** [comment]
welcome to trying to use tfc and greg with other mods stuff isnt going to work right and the mod devs dont help lol

> Okay, didn’t know it was a mod-specific bug.
> 
> Personally, this is IMO the best backpack mod going around & the bug is purely cosmetic so I can live with it, I request we don’t change it.
> 
> thanks for replying!
welcome to trying to use tfc and greg with other mods stuff isnt going to work right and the devs for these mods dont help lol

**Comment by BabBimmaBome** [comment]
https://www.curseforge.com/minecraft/mc-mods/useful-backpacks maybe this oldy but a goodie would work?

**Comment by Exzept1on** [comment]
Yes, it was in the direction of this mod that we looked, we also considered backpacks from crayfish, but have not yet decided.

**Comment by Exzept1on** [comment]
> Okay, didn’t know it was a mod-specific bug.
> 
> Personally, this is IMO the best backpack mod going around & the bug is purely cosmetic so I can live with it, I request we don’t change it.
> 
> thanks for replying!

that true, it best but this...

**Comment by chemlzh** [comment]
@Exzept1on I think it's because you have banned original recipes of sophisticated backpacks, which include dyeing recipes for backpacks. These recipes are hard-coded in https://github.com/P3pp3rF1y/SophisticatedBackpacks/blob/1.20.x/src/main/java/net/p3pp3rf1y/sophisticatedbackpacks/crafting/BackpackDyeRecipe.java

**Comment by chemlzh** [comment]
We'd better figure out how to bring back dyeing recipes back or ban other recipes without interfering dyeing.

**Comment by chemlzh** [comment]
I have found a solution, and I'll push it later.
